apps/chat/bots.py

```diff
@@ -287,3 +287,13 @@ def process_input(self, user_input: str, save_input_to_history=True, attachments
             PipelineState(messages=[user_input], experiment_session=self.session), self.session
         )
         return output["messages"][-1]
+
+    def get_ai_message_id(self) -> int | None:
```

Comment:
Not sure if this is the right way of going about this... Currently we always save history for a pipeline run, but that might not always be true. I guess in that case this would be None, and we move on as before 🤷🏽‍♂️ . However, this does feel prone to race conditions if this is fetched for some reason before the message was committed...

Reply:
A more robust way would be for the pipeline to return the AI generated `ChatMessage`. I am doing some work in this area at the moment for versioning, so I'll update the code accordingly. I think this approach is fine for now. It works